app/models/reservation.rb

```diff
@@ -14,6 +14,9 @@ class Reservation < ActiveRecord::Base
     record.errors.add(attr, 'cannot be a guest') if value.role == 'guest'
   end
   validate :start_date_before_due_date
+  validate :check_out_time_after_start_date
```

Comment:
I think we can probably consolidate these three validations into a single validation (more comments below).